### PR TITLE
Use enum type for Job.Action, Heartbeat.Action and ComponentSpec.Action

### PR DESCRIPTION
**Pull request by RemindD**
fix https://github.com/eclipse-symphony/symphony/issues/65

Use enum type for three Action field so that we don't need to worry about case sensitive or not.

### CODE REVIEW

File: api/pkg/apis/v1alpha1/providers/stage/create/create.go [hunk 1]
```diff
@@ -139,12 +140,12 @@ func (i *CreateStageProvider) Process(ctx context.Context, mgrContext contexts.M
 			objectNamespace = "default"
 		}
 
-		if action == "remove" {
+		if strings.EqualFold(action, "remove") {
```
Comment (Haishi2016) [inline api/pkg/apis/v1alpha1/providers/stage/create/create.go:143]: do we want to compare with the constant here as well?

File: api/pkg/apis/v1alpha1/providers/target/mock/mock.go [hunk 2]
```diff
@@ -124,7 +124,7 @@ func (m *MockTargetProvider) Apply(ctx context.Context, deployment model.Deploym
 		for i, _ := range cache[m.Config.ID] {
 			if cache[m.Config.ID][i].Name == c.Component.Name {
 				found = true
-				if c.Action == "delete" {
+				if c.Action == model.ComponentDelete {
```
Comment (Haishi2016) [inline api/pkg/apis/v1alpha1/providers/target/mock/mock.go:127]: This is inconsistent. Whether we do case-sensitive comparison everywhere, or we do case-insensitive comparison everywhere

File: api/pkg/apis/v1alpha1/providers/stage/create/create.go (hunk 1 above)
Comment (RemindD) [reply api/pkg/apis/v1alpha1/providers/stage/create/create.go:143]: This is passed in from user's yaml file. We cannot guarantee it's well formatted.

File: api/pkg/apis/v1alpha1/providers/target/mock/mock.go (hunk 2 above)
Comment (RemindD) [reply api/pkg/apis/v1alpha1/providers/target/mock/mock.go:127]: In my opinion, for internal state, we need to use enum values and use case sensitive comparison with enum values, while for inputs from users. we need to use case insensitive comparison for better user experience.

File: api/pkg/apis/v1alpha1/providers/stage/create/create.go (hunk 1 above)
Comment (Haishi2016) [reply api/pkg/apis/v1alpha1/providers/stage/create/create.go:143]: I mean to use strings.EqualFold(action, <the constant>)

Comment (RemindD) [reply api/pkg/apis/v1alpha1/providers/stage/create/create.go]: Fixed.

**Review by Haishi2016 — APPROVED** [review]

**Pull request merged**